openfeature/client.go

```diff
@@ -32,20 +32,20 @@ type IClient interface {
 
 // ClientMetadata provides a client's metadata
 type ClientMetadata struct {
-	name string
+	domain string
 }
 
 // NewClientMetadata constructs ClientMetadata
 // Allows for simplified hook test cases while maintaining immutability
-func NewClientMetadata(name string) ClientMetadata {
+func NewClientMetadata(domain string) ClientMetadata {
 	return ClientMetadata{
-		name: name,
+		domain: domain,
 	}
 }
 
-// Name returns the client's name
-func (cm ClientMetadata) Name() string {
-	return cm.name
+// Domain returns the client's domain
+func (cm ClientMetadata) Domain() string {
+	return cm.domain
 }
```

Comment:
We should avoid this breaking change. 

Instead, keep the old method and make it deprecate (use comment `// Deprecated` [1]). And add the new method with `Domain() string` signature, returning the `cm.domain`

[1] - https://go.dev/wiki/Deprecated